plugins/org.eclipse.xtext/src/org/eclipse/xtext/generator/OutputConfigurationProvider.java

```diff
@@ -40,6 +49,12 @@
 	 */
 	@Override
 	public Set<OutputConfiguration> getOutputConfigurations(Resource context) {
-		return getOutputConfigurations();
+		EList<Adapter> adapters = context.getResourceSet().eAdapters();
+		OutputConfigurationAdapter adapter = (OutputConfigurationAdapter) EcoreUtil.getAdapter(adapters, OutputConfigurationAdapter.class);
+		if (adapter == null) {
+			return getOutputConfigurations();
+		} else {
+			return adapter.getOutputConfigurationsPerLanguage().get(languageName);
```

Comment:
Should the provider init the adapter if the language wasn't contained in there, so far?